client/src/mapper-consents/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { apiErrorMessage, getMapperConsents, useApi } from '../api';
+import { BeatmapInline } from '../BeatmapInline';
+import { IBeatmapset, IMapperBeatmapsetConsent, IMapperConsent } from '../interfaces';
+import { UserInline } from '../UserInline';
+
+interface IMapperBeatmapsetConsentGrouped extends Omit<IMapperBeatmapsetConsent, 'beatmapset_id'> {
+  beatmapset: IBeatmapset
+}
+
+interface IMapperConsentGrouped extends Omit<IMapperConsent, 'beatmapset_consent'> {
+  beatmapset_consents: IMapperBeatmapsetConsentGrouped[]
+}
+
+function consentToCell(consent?: 0 | 1 | 2 | boolean) {
+  let index: 'null' | 0 | 1 | 2;
+  if (consent === true) {
+    index = 1
+  } else if (consent === false) {
+    index = 0
+  } else {
+    index = consent == null ? 'null' : consent;
+  }
+  const mapping = {
+    'null': ['no reply', 'pending'],
+    0: ['no', 'error'],
+    1: ['yes', 'success'],
+    2: ['unreachable', 'pending']
+  };
+  const [consentStr, className] = mapping[index];
+
+  return (
+    <td className={className}>
+      {consentStr}
+    </td>
+  )
+}
+
+function maybeBeatmapsetTable(consent: IMapperConsentGrouped) {
+  if (consent.beatmapset_consents.length === 0) {
+    return
+  }
+  return (
+    <tr key={consent.id + "-dropdown"}>
+      <td colSpan={3}>{MapperBeatmapsetConsents(consent)}</td>
+    </tr>
+  )
+}
+
+function MapperBeatmapsetConsents(mapperConsent: IMapperConsentGrouped) {
+  return (
+    <table style={{'width': '80%', 'marginLeft': '3em', 'marginRight': '3em', 'tableLayout': 'fixed'}}>
+      <tr>
+          <th style={{'width': '30%'}}>Beatmapset</th>
+          <th style={{'width': '10%'}}>Consent</th>
+          <th style={{'width': '60%'}}>Notes</th>
+      </tr>
+      {mapperConsent.beatmapset_consents.map((consent) => {
+        // TODO remove when beatmapset foreign key issues have been resolved
+        if (consent.beatmapset == null) {
```

Comment:
as you mentioned on discord, foreign key constraints are not being respected right now, so this is a temporary null check until they are (the page will also look not great until they are)